### ISSUE TITLE
azure: get away from password auth

### ISSUE THREAD

**Issue by colemickens** [kubernetes/kubernetes-anywhere#182]
@errordeveloper / @mikedanese How is auth handled for GCE/AWS?

I don't like using password auth for the VMs. I'm guessing GCE just deploys without auth and expects you to use `gcloud` or Cloud Console to push short lived keys into the VM just-in-time?

I'm very surprised there isn't an SSH provisioner for Terraform...

**Comment by errordeveloper** [comment]
In EC2 you have an SSH key that you pick for your instances, it is the job
of the OS provisioning scripts (e.g. cloud-init) to put that in the right
place. It depend on the distro which user you will get, on Ubuntu it is the
'ubuntu' user, and on Amazon Linux it is the 'ec2-user.  You can configure
your instances without SSH too, but EC2 doesn't provide any other means on
debugging issue, so I doubt anyone does that. You can either import a
public key, or generate a key pair for you. However, there is only one
public key that can be associated with an instance. The GCE has a much
nicer solution to all this, with dynamic user provisioning, but it requires
an agent. I forgot how exactly default user account work though... I
haven't used Azure with Terraform, so I am not sure what is the issue
exactly. Is it somehow to do with different Azure account types?

**Comment by colemickens** [comment]
Yes, user `authorized_key` provisioning works the same or similarly in Azure, but at the moment I'm only using password auth currently with Terraform.

Is the Terraform/aws/phase1 implementation using SSH keys? Is it a required field in the `config.json` or otherwise how is it specified as input to the terraform template?

**Comment by errordeveloper** [comment]
If a key name is unspecified, current implementation actually expects
'kubernetes-anywhere' key to exist. User can import the key using
additional 'aws_key_pair', but that's something I discovered recently, so I
haven't documented it yet. I would be very glad if we could align on where
SSH keys should come from, and make the change once we have consensus.

Now I recall importing my key to GCE as metadata, when I used it with
Terraform, but I forgot the exact context, and whether I only had to do it
once for my account?
